### PR TITLE
Export History

### PR DESCRIPTION
**Pull request by YunchuWang**
# Summary

## What changed?

Add a new `@microsoft/durabletask-js-export-history` package that enables exporting orchestration instance history to Azure Blob Storage. This is a JavaScript/TypeScript port of the Export History feature from `durabletask-dotnet`.

**New package: `packages/durabletask-js-export-history/`**

- **Entity** (`export-job.ts`): Durable entity that tracks export job lifecycle (Pending → Active → Completed/Failed) with checkpoint support and state transitions
- **Orchestrators**:
  - `export-job-orchestrator.ts`: Main orchestrator that lists terminal instances, exports their history in batches with retry logic (exponential backoff), commits checkpoints, and marks completion/failure
  - `execute-export-job-operation-orchestrator.ts`: Wrapper orchestrator that creates the entity and starts the export job
- **Activities**:
  - `export-instance-history-activity.ts`: Fetches instance history via gRPC, serializes to JSONL/JSON format, compresses with gzip, and uploads to Azure Blob Storage
  - `list-terminal-instances-activity.ts`: Lists orchestration instances in terminal states within a time window
- **Client** (`export-history-client.ts`): `ExportHistoryClient` and `ExportHistoryJobClient` classes for creating, querying, deleting, and describing export jobs
- **Builders** (`use-export-history.ts`): `useExportHistoryWorker()` helper to register all export history orchestrators, activities, and entities on a worker
- **Models**: Full set of TypeScript types for export configuration, job state, destinations, formats, filters, checkpoints, and failures
- **Errors**: Custom error classes (`ExportJobClientValidationError`, `ExportJobInvalidTransitionError`)
- **Constants**: Named constants for orchestrators, activities, and entities

**Test coverage:**

- 54 unit tests across 6 test suites (builders, constants, entity, errors, models, list-terminal-instances-activity)
- 18 E2E tests (`test/e2e-azuremanaged/export-history.spec.ts`) covering:
  - Entity lifecycle (create, query, delete, auto-generated jobId)
  - Export execution & blob verification (N-instance export, empty window, JSONL.GZ content, JSON format, multi-step orchestrations, blob isolation, failed orchestrations, mixed status, full history event sequence matching)
  - Configuration validation (defaults, missing connectionString/containerName)
  - Diagnostics (orchestrator start verification)
- E2E tests are gated behind `EXPORT_HISTORY_E2E=1` env var (skipped in CI — DTS emulator lacks entity support)

**Other changes:**

- `jest.config.js`: Added module name mapping for `@microsoft/durabletask-js-export-history`
- `CHANGELOG.md`: Removed duplicate v0.2.0 entries (branch was based on pre-v0.2.0)
- Version alignment: `durabletask-js` and `durabletask-js-azuremanaged` package versions set to `0.1.0-beta.1`

## Why is this change needed?

Enables users to export durable orchestration instance history to Azure Blob Storage for auditing, analytics, and long-term retention. This fills a gap in the JS SDK where the equivalent feature already exists in the .NET SDK (`durabletask-dotnet`).

## Issues / work items

- Related: Export History feature parity with `durabletask-dotnet`

---

# Project checklist

- [ ] Release notes are not required for the next release
  - [ ] Otherwise: Notes added to `CHANGELOG.md`
- [ ] Backport is not required
  - [ ] Otherwise: Backport tracked by issue/PR #issue_or_pr
- [x] All required tests have been added/updated (unit tests, E2E tests)
- [ ] Breaking change?
  - No breaking changes. This is a new additive package.

---

# AI-assisted code disclosure (required)

## Was an AI tool used? (select one)

- [ ] No
- [ ] Yes, AI helped write parts of this PR (e.g., GitHub Copilot)
- [x] Yes, an AI agent generated most of this PR

If AI was used:
- Tool(s): GitHub Copilot (Claude)
- AI-assisted areas/files: All files in `packages/durabletask-js-export-history/`, `test/e2e-azuremanaged/export-history.spec.ts`, `jest.config.js`
- What you changed after AI output: Reviewed and iterated on orchestrator logic, entity state transitions, error handling, and E2E test correctness

AI verification (required if AI was used):
- [x] I understand the code and can explain it
- [x] I verified referenced APIs/types exist and are correct
- [x] I reviewed edge cases/failure paths (timeouts, retries, cancellation, exceptions)
- [x] I reviewed concurrency/async behavior
- [x] I checked for unintended breaking or behavior changes

---

# Testing

## Automated tests

- Unit tests: 54 passed, 0 failed (6 test suites)
- E2E tests: 18 passed, 0 failed (requires `EXPORT_HISTORY_E2E=1` + real DTS scheduler + Azurite)

## Manual validation (only if runtime/behavior changed)

- Environment: Windows 10, Node.js 24.12.0, DTS Scheduler (West US 3), Azurite 3.35.0
- Steps + observed results:
  1. Started Azurite locally, configured DTS connection string
  2. Ran all 18 E2E tests against real DTS scheduler — all passed (239s total)
  3. Verified exported blobs in Azurite contain correct JSONL.GZ history events matching orchestration execution sequence
- Evidence: Full history event sequence verified (OrchestratorStarted → ExecutionStarted → TaskScheduled → OrchestratorStarted → TaskCompleted → ExecutionCompleted)

---

# Notes for reviewers

- The E2E tests require a real DTS scheduler with entity support. They are gated behind `EXPORT_HISTORY_E2E=1` and will auto-skip in CI.
- The `createTimer` calls in the JS orchestrator pass `new Date()` objects (not raw numbers) because the JS SDK interprets numeric arguments as seconds-delay and can produce invalid timestamps.
- The JS SDK does not support `RetryPolicy` on `callActivity` due to a DTS server-side Timestamp validation bug, so retry is handled at the batch level in the orchestrator.
- Version numbers were aligned to `0.1.0-beta.1` to match the branch base. The CHANGELOG removal of v0.2.0 entries should be reverted if this PR targets post-v0.2.0.

### CODE REVIEW

File: packages/durabletask-js-export-history/src/client/export-history-client.ts [hunk 1]
```diff
@@ -0,0 +1,273 @@
...
+    const state = await this.client.waitForOrchestrationCompletion(instanceId, true, 120);
+
+    if (!state || state.runtimeStatus !== OrchestrationStatus.COMPLETED) {
+      throw new Error(
+        `Failed to create export job '${this.jobId}': ${state?.failureDetails?.message ?? "Unknown error"}`,
+      );
+    }
+
+    // The entity's Create operation signals Run, which schedules the export orchestrator.
+    // No need to schedule the orchestrator from the client.
```
Comment (halspang) [inline packages/durabletask-js-export-history/src/client/export-history-client.ts]: Why the comments at the end here?

File: packages/durabletask-js-export-history/src/models/export-job-status.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+export enum ExportJobStatus {
+  /**
+   * The export job has not been started.
+   */
+  Pending = "Pending",
+
+  /**
+   * The export job is actively running.
+   */
+  Active = "Active",
```
Comment (halspang) [inline packages/durabletask-js-export-history/src/models/export-job-status.ts]: `Active` -> `Running`?

File: packages/durabletask-js-export-history/src/orchestrators/execute-export-job-operation-orchestrator.ts [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+import { OrchestrationContext, TOrchestrator, EntityInstanceId } from "@microsoft/durabletask-js";
+
+/**
+ * Request to execute an operation on an ExportJob entity via orchestration.
+ *
+ * Note: `entityId` is serialized as a string (e.g. "@exportjob@my-key") rather than
+ * an EntityInstanceId object to ensure it survives JSON round-tripping through
+ * orchestration input serialization.
+ */
+export interface ExportJobOperationRequest {
```
Comment (halspang) [inline packages/durabletask-js-export-history/src/orchestrators/execute-export-job-operation-orchestrator.ts]: Nit: Since we have a model directory, any reason why we didn't include this (and a few other request objects)?

File: packages/durabletask-js-export-history/src/errors.ts [hunk 4]
```diff
@@ -0,0 +1,50 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
```
Comment (halspang) [inline packages/durabletask-js-export-history/src/errors.ts]: I would love an illegal argument style error. We have a lot of input validation that just throws a generic error.

File: packages/durabletask-js-export-history/package.json [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+    ".": {
+      "types": "./dist/index.d.ts",
+      "require": "./dist/index.js",
+      "import": "./dist/index.js"
+    }
+  },
+  "files": [
+    "dist",
+    "LICENSE"
+  ],
```
Comment (halspang) [inline packages/durabletask-js-export-history/package.json]: Do these represent real files? Cause I don't see any of that name, or does it exist elsewhere in the repo (outside of this PR)?
